src/v/cluster/cluster_discovery.cc

```diff
@@ -148,18 +162,185 @@ ss::future<bool> cluster_discovery::dispatch_node_uuid_registration_to_seeds(
     co_return false;
 }
 
-/*static*/ std::optional<node_id>
+ss::future<cluster_bootstrap_info_reply>
+cluster_discovery::request_cluster_bootstrap_info_single(
+  const net::unresolved_address addr) const {
+    vlog(clusterlog.info, "Requesting cluster bootstrap info from {}", addr);
+    _as.check();
+    cluster_bootstrap_info_reply reply;
+    for (auto repeat_jitter = simple_time_jitter<model::timeout_clock>(1s);;) {
+        result<cluster_bootstrap_info_reply> reply_result(
+          std::errc::connection_refused);
+        try {
+            reply_result = co_await do_with_client_one_shot<
+              cluster_bootstrap_client_protocol>(
+              addr,
+              config::node().rpc_server_tls(),
+              2s,
+              [](cluster_bootstrap_client_protocol c) {
+                  return c
+                    .cluster_bootstrap_info(
+                      cluster_bootstrap_info_request{},
+                      rpc::client_opts(rpc::clock_type::now() + 2s))
+                    .then(&rpc::get_ctx_data<cluster_bootstrap_info_reply>);
+              });
+            if (reply_result) {
+                reply = std::move(reply_result.value());
+                break;
+            }
+        } catch (...) {
+            vlog(
+              clusterlog.warn,
+              "Error requesting cluster bootstrap info from {}, retrying. {}",
+              addr,
+              std::current_exception());
+        }
+        co_await ss::sleep_abortable(repeat_jitter.next_duration(), _as);
+        vlog(clusterlog.trace, "Retrying cluster bootstrap info from {}", addr);
+    }
+
+    vlog(clusterlog.info, "Obtained cluster bootstrap info from {}", addr);
+    vlog(clusterlog.debug, "{}", reply);
+    co_return std::move(reply);
+}
+
+ss::future<> cluster_discovery::fetch_cluster_bootstrap_info() {
+    if (_cluster_bootstrap_info_fetched) co_return;
+    const auto units = co_await ss::get_units(
+      _fetch_cluster_bootstrap_sem, 1, _as);
+    if (_cluster_bootstrap_info_fetched) co_return;
+
+    const net::unresolved_address& self_addr
+      = config::node().advertised_rpc_api();
+    const std::vector<config::seed_server>& self_seed_servers
+      = config::node().seed_servers();
+
+    std::vector<
+      std::pair<net::unresolved_address, cluster_bootstrap_info_reply>>
+      peers;
+    peers.reserve(self_seed_servers.size());
+    for (const config::seed_server& seed_server : self_seed_servers) {
+        // do not call oneself
+        if (seed_server.addr == self_addr) continue;
+        peers.emplace_back(seed_server.addr, cluster_bootstrap_info_reply{});
+    }
+    co_await ss::parallel_for_each(peers, [this](auto& peer) -> ss::future<> {
+        peer.second = co_await request_cluster_bootstrap_info_single(
+          peer.first);
```

Comment:
Are we guaranteed that this returns a reply, e.g. if the seed is down?

Just want to make sure we're able to get through here when at least one seed is down.

Reply:
Actually it won't, and your point is correct, this shouldn't be the case for wiped seed nodes. I will attend to this in a subsequent PR.